client/jetpack-onboarding/main.jsx

```diff
@@ -27,22 +29,43 @@ class JetpackOnboardingMain extends React.PureComponent {
 		stepName: STEPS.SITE_TITLE,
 	};
 
-	// TODO: Add lifecycle methods to redirect if no siteId
+	componentDidMount() {
+		const { siteId, siteSlug } = this.props;
+
+		// If we are missing the Jetpack onboarding credentials,
+		// redirect back to wp-admin so we can obtain them again.
+		if ( ! siteId && siteSlug ) {
+			const siteDomain = siteSlug.replace( '::', '/' );
+			const url = addQueryArgs(
+				{
+					page: 'jetpack',
+					action: 'onboard',
+					calypso_env: config( 'env_id' ),
+				},
+				`//${ siteDomain }/wp-admin/admin.php`
+			);
+			externalRedirect( url );
```

Comment:
I'm wondering if we should try accessing the Jetpack site before redirecting to it (similar to how JPC checks it), and if it's not a valid WordPress site with Jetpack installed and activated, redirect to `/` or something...

Reply:
Could be material for another iteration :slightly_smiling_face: